project/src/pages/favorites/favorites.tsx

```diff
@@ -1,134 +1,60 @@
 import { Link } from 'react-router-dom';
 import Layout from '../../components/layout/layout';
+import { Offer } from '../../types/offers';
+import Offers from '../../components/offers/offers';
+
+type FavoritesProps = {
+  offers: Offer[];
+}
+type OffersByCity = {
+  [cityName: string]: Offer[];
+}
+
+
+function FavoritesPage({ offers }: FavoritesProps): JSX.Element {
+
+  function filterByIsFavorite(allOffers: Offer[]): Offer[] {
+
+    return allOffers.filter((offer) => offer.isFavorite);
+  }
+  const favoriteOffers = filterByIsFavorite(offers);
+
+  function groupOffersByCity(allOffers: Offer[]): OffersByCity {
+
+    return allOffers.reduce((offersByCity: OffersByCity, offer) => {
+      const { city: { name: cityName } } = offer;
+      if (!offersByCity[cityName]) {
+        offersByCity[cityName] = [];
+      }
+      offersByCity[cityName].push(offer);
+      return offersByCity;
+    }, {});
```

Comment:
Супер, я бы только форматирования добавил немного
```suggestion
      const { city: { name: cityName } } = offer;
      
      if (!offersByCity[cityName]) {
        offersByCity[cityName] = [];
      }
      
      offersByCity[cityName].push(offer);
      
      return offersByCity;
    }, {});
```